imagebuildah/build.go

```diff
@@ -0,0 +1,331 @@
+	"github.com/docker/docker/builder/dockerfile/parser"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/openshift/imagebuilder"
+	"github.com/projectatomic/buildah"
+)
+
+// BuildDockerfileOptions can be used to alter how an image is built.
+type BuildDockerfileOptions struct {
+	// ContextDirectory is the default source location for COPY and ADD
+	// commands.
+	ContextDirectory string
+	// PullIfMissing signals to NewBuilder() that it should pull the image
+	// if it is not present in the local Store.
+	PullIfMissing bool
+	// PullAlways signals to NewBuilder() that it should pull the source
+	// image before creating the container.
+	PullAlways bool
+	// Registry is a value which is prepended to the image's name, if it
+	// needs to be pulled and the image name alone can not be resolved to a
+	// reference to a source image.
+	Registry string
+	// IgnoreUnrecognizedInstructions tells us to just log instructions we
+	// don't recognize, and try to keep going.
+	IgnoreUnrecognizedInstructions bool
+	// Quiet tells us whether or not to announce steps as we go through them.
+	Quiet bool
+	// Runtime is the name of the command to run for RUN instructions.  It
+	// should accept the same arguments and flags that runc does.
+	Runtime string
+	// RuntimeArgs adds global arguments for the runtime.
+	RuntimeArgs []string
+	// TransientMounts is a list of mounts that won't be kept in the image.
+	TransientMounts []specs.Mount
+	// Compression specifies the type of compression which is applied to
+	// layer blobs.  The default is to not use compression, but
+	// archive.Gzip is recommended.
+	Compression archive.Compression
+	// Arguments which can be interpolated into Dockerfiles
+	Args map[string]string
+	// Name of the image to write to.
+	Output string
+	// SignaturePolicyPath specifies an override location for the signature
+	// policy which should be used for verifying the new image as it is
+	// being written.  Except in specific circumstances, no value should be
+	// specified, indicating that the shared, system-wide default policy
+	// should be used.
+	SignaturePolicyPath string
+}
+
+type buildahExecutor struct {
+	store                          storage.Store
+	contextDir                     string
+	builder                        *buildah.Builder
+	pullIfMissing                  bool
+	pullAlways                     bool
+	registry                       string
+	ignoreUnrecognizedInstructions bool
+	counter                        int
+	quiet                          bool
+	runtime                        string
+	runtimeArgs                    []string
+	transientMounts                []specs.Mount
+	compression                    archive.Compression
+	output                         string
+	signaturePolicyPath            string
+	systemContext                  *types.SystemContext
+}
+
+func getSystemContext(signaturePolicyPath string) *types.SystemContext {
```

Comment:
Here is the suggested style reference: https://golang.org/doc/effective_go.html#Getters